docs/guides/go-built-in.md

```diff
@@ -462,16 +460,33 @@ execution of the block once accepted.
 In the following code, the application simply returns the unmodified group of transactions:
 
 ```go
-func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
-    return &abcitypes.ResponsePrepareProposal{Txs: proposal.Txs}, nil
-}
-```
+ func (app *KVStoreApplication) PrepareProposal(_ context.Context, proposal *abcitypes.RequestPrepareProposal) (*abcitypes.ResponsePrepareProposal, error) {
+   totalBytes := int64(0)
+   txs := make([]byte, 0)
+
+   for _, tx := range proposal.Txs {
+     totalBytes += int64(len(tx))
+     txs = append(txs, tx...)
 
-Once a proposed block is received by a node, the proposal is passed to the application to give
-its blessing before voting to accept the proposal.
+       if totalBytes > int64(proposal.MaxTxBytes) {
```

Comment:
The use of totalBytes is redundant and could instead just use len(txs) but the even better check would be to firstly check if the final added value is greater than proposal.MaxTxBytes

```go
txs := make([]byte, 0, len(proposal.Txs)*100)
max := int64(proposal.MaxTxBytes)

for _, tx := range proposal.Txs {
   if n := len(txs) + len(tx); n > max || n <= 0 {
       break
   }
```

Reply:
Indeed, @odeke-em Thanks for pointing it out.
Given that this PR is now merged, would you mind opening a new PR with your suggestions?

Reply:
Isn't the problem here that the `txs` array is not of the right type? (is `[]byte`, should be `[][]byte`)
Then we are "coalescing" all transactions into a byte array... but that's not what we should be doing, is it?

Reply:
You are right, @sergio-mena, but @odeke-em has a point in that the overflow check should be done before the last transaction has been added, otherwise it is too late to break the loop.